syntax = "proto3";

package book;

service InventoryService{
  // A simple RPC.
  //
  // Obtains the MessageResponse at a given position.
 rpc Create_Book(Book) returns (Create_book_reply) {}

 //
 rpc Get_Book(Book) returns (Book) {}

}

message Book_reply{
    string ISBN = 1;
}

message Create_book_reply {
    string success = 1;
}

message Book{
 string ISBN = 1;
 string title = 2;
 string author = 3;
 int32 publishing_year = 4;
 enum Genre {
    SCIFI = 0;
    HORROR = 1;
    ROMANCE = 2;
    FANTASY = 3;
 }
}

message Inventory_Item{
 int32 inventory_number = 5;
 oneof Book{
    string ISBN = 1;
    string title = 2;
    string author = 3;
    int32 publishing_year = 4;
    }
 enum STATUS {
    AVAILABLE = 0;
    TAKEN = 1;
 }
}